### DISCUSSION TITLE
Issues downloading foundry

### DISCUSSION THREAD

**Discussion by infinite969** [Cyfrin/foundry-full-course-cu#1068] [Q&A]
Hello friends, I am having an issue downloading items in foundry after commanding foundryup in my VScode terminal here is my code

base) cooperpowell@Coopers-MacBook-Pro ~ % foundryup

.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx
 
 ╔═╗ ╔═╗ ╦ ╦ ╔╗╔ ╔╦╗ ╦═╗ ╦ ╦         Portable and modular toolkit
 ╠╣  ║ ║ ║ ║ ║║║  ║║ ╠╦╝ ╚╦╝    for Ethereum Application Development 
 ╚   ╚═╝ ╚═╝ ╝╚╝ ═╩╝ ╩╚═  ╩                 written in Rust.

.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx

Repo       : https://github.com/foundry-rs/
Book       : https://book.getfoundry.sh/                      
Chat       : https://t.me/foundry_rs/                         
Support    : https://t.me/foundry_support/
Contribute : https://github.com/orgs/foundry-rs/projects/2/

.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx

foundryup: installing foundry (version nightly, tag nightly)
foundryup: downloading latest forge, cast, anvil, and chisel
########################################################################################## 100.0%
foundryup: downloading manpages
########################################################################################## 100.0%
dyld: Library not loaded: /usr/local/opt/libusb/lib/libusb-1.0.0.dylib
  Referenced from: /Users/cooperpowell/.foundry/bin/forge
  Reason: image not found
foundryup: command failed: /Users/cooperpowell/.foundry/bin/forge --version
foundryup: installed - 
dyld: Library not loaded: /usr/local/opt/libusb/lib/libusb-1.0.0.dylib
  Referenced from: /Users/cooperpowell/.foundry/bin/cast
  Reason: image not found
foundryup: command failed: /Users/cooperpowell/.foundry/bin/cast --version
foundryup: installed - 
dyld: Library not loaded: /usr/local/opt/libusb/lib/libusb-1.0.0.dylib
  Referenced from: /Users/cooperpowell/.foundry/bin/anvil
  Reason: image not found
foundryup: command failed: /Users/cooperpowell/.foundry/bin/anvil --version
foundryup: installed - 
dyld: Library not loaded: /usr/local/opt/libusb/lib/libusb-1.0.0.dylib
  Referenced from: /Users/cooperpowell/.foundry/bin/chisel
  Reason: image not found
foundryup: command failed: /Users/cooperpowell/.foundry/bin/chisel --version
foundryup: installed - 
foundryup: done!
(base) cooperpowell@Coopers-MacBook-Pro ~ %

**Comment by infinite969** [comment]
I seem to not be able to load forge, cast, anvil, chisel. If someone can help me troubleshoot that would be awesome!

**Reply by Kuusho** [reply]
Try going to the directory where your foundry is downloaded and deleting with `rm -r foundry/` then reinstalling it again from your regular terminal not the one in VScode and running `foundryup` once that's done.

**Comment by infinite969** [comment]
base) ***@***.*** ~ % rm -r foundry/
rm: foundry/: No such file or directory
(base) ***@***.*** ~ % rm -r foundry
rm: foundry: No such file or directory

getting this error when entering into the VScode terminaL

**Reply by Kuusho** [reply]
use the command when you get to the directory where you downloaded foundry.

from your current directory use `ll` to list everything in it. if `.foundry/` is there, then use `rm -r .foundry/` then continue to download foundry again with `curl -L https://foundry.paradigm.xyz | bash`

**Comment by Booker-CU** [comment]
I've located the .foundry/ file and removed it and re-installed foundry, but when I open a new terminal and do 'foundryup' it still says command not found : ( 
--it seems very particular* on mac or something or at least mine

**Reply by Jamster187** [reply]
hey booker,

I had this same problem.

try this:

in terminal:

```
cd 
cd /Users/YOURUSERNAMEHERE/
code .bash_profile (you'll need vscode code extension to be installed - https://code.visualstudio.com/docs/setup/mac)
```

add this in your .bash_profile somewhere:

```
export PATH="$PATH:/Users/YOURUSERNAMEHERE/.foundry/bin"
export PATH="$PATH:/Users/YOURUSERNAMEHERE/.local/bin"
```

restart terminal and confirm success by typing this in terminal:

```
forge --version
```

hope this helps

**Reply by pathakadi** [reply]
thanks @Jamster187, It worked.

**Comment by AbigirlMuchineripi** [comment]
Hello, @infinite969 did you find the solution ?
